agora/Agora/Effect/GovernorMutation.hs

```diff
@@ -140,62 +138,49 @@ instance PTryFrom PData (PAsData PMutateGovernorDatum) where
 -}
 mutateGovernorValidator :: Governor -> ClosedTerm PValidator
 mutateGovernorValidator gov = makeEffect (authorityTokenSymbolFromGovernor gov) $
-  \_gatCs (datum' :: Term _ PMutateGovernorDatum) txOutRef txInfo' -> P.do
+  \_gatCs (datum' :: Term _ PMutateGovernorDatum) _ txInfo' -> P.do
     datum <- pletFields @'["newDatum", "governorRef"] datum'
-
     txInfo <- pletFields @'["mint", "inputs", "outputs", "datums"] txInfo'
 
-    let selfAddress =
-          pfield @"address"
-            #$ mustBePJust # "Self input not found"
-            #$ findTxOutByTxOutRef # txOutRef # txInfo.inputs
-
-    passert "No output to the effect validator" $
-      pnull #$ findOutputsToAddress # txInfo.outputs # selfAddress
-
     let mint :: Term _ (PBuiltinList _)
         mint = pto $ pto $ pto $ pfromData txInfo.mint
 
     passert "Nothing should be minted/burnt other than GAT" $
       plength # mint #== 1
 
-    filteredInputs <-
-      plet $
-        pfilter
-          # plam
-            ( \inInfo ->
-                let value = pfield @"value" #$ pfield @"resolved" # inInfo
-                 in gstValueOf # value #== 1
-            )
-          # pfromData txInfo.inputs
+    passert "Only self and governor inputs are allowed" $
+      plength # pfromData txInfo.inputs #== 2
```

Comment:
For inputs, you don't want to simple restriction with number of inputs because it includes both script input and user input. To run effect as a third party, other users should be able to pay the fees. So, I think, you can limit the number of `ScriptCredential` to 2 but allow other inputs from `PubKeyCredential`. Sorry for phrasing it little ambiguously on the first comment : )

For outputs, limiting to only one output is working great with this effect.

For the reference check [this out.](https://github.com/Liqwid-Labs/agora/pull/56#discussion_r857539581)

@emiflake